### PR TITLE
REALMC-7564 map ordering when parsing a bson.D

### PR DESCRIPTION
**Pull request by sudssm**
preserve order from bson.D

### CODE REVIEW

File: schema.go [hunk 1]
```diff
@@ -1105,19 +1105,31 @@ func (d *Schema) parseReference(documentNode interface{}, currentSchema *subSche
...
+		return nil
+	}
 
 	m, err := mustBeMap(documentNode)
 	if err != nil {
 		return err
 	}
 
 	for k := range m {
-		newSchema := NewSubSchema(k, currentSchema)
```
Comment (tkaye407) [inline schema.go:1116]: what a strange loop they have here

**Review by tkaye407 — APPROVED** [review]
Good stuff, like the optimistic check for bson.D 👍

**Comment by tsedgwick** [comment]
before merging, can we update the title with the jira ticket and an imperative sentence please?

**Review by mpobrien — APPROVED** [review]
👏

**Pull request merged**